compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1599,4 +1600,32 @@ impl<'tcx> LifetimeContext<'_, 'tcx> {
             _ => {}
         }
     }
+
+    /// Non-static lifetimes are prohibited in anonymous constants under `min_const_generics` so
+    /// this function will emit an error if `min_const_generics` is enabled, the body identified by
+    /// `body_id` is a anonymous constant and `lifetime_ref` is non-static.
+    crate fn maybe_emit_forbidden_non_static_lifetime_error(
+        &self,
+        body_id: hir::BodyId,
+        lifetime_ref: &'tcx hir::Lifetime,
+    ) {
+        let is_anon_const = matches!(
+            self.tcx.def_kind(self.tcx.hir().body_owner_def_id(body_id)),
```

Comment:
how does this deal with
```rust
fn test<'a>() -> [u8; {
    |()| {
        let _: &'a () = &();   
    };
    7
}] { todo!() }
```
It looks to me like this always causes an ICE, even with `#[feature(const_generics)]`

Reply:
This causes triggers the error with this patch and `#![feature(min_const_generics)]` and an ICE with this patch and `#![feature(const_generics)]`. It's a different ICE than before though:

```
thread 'rustc' panicked at 'escaping bound vars for WellFormed(&'a ())', /home/david/Projects/rust/rust8/compiler/rustc_middle/src/ty/mod.rs:1396:9
```

Reply:
Added this to the test.